support/package_nginx

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+#
+# This script is used for the [Nginx
+# buildpack](github.com/Scalingo/nginx-buildpack/). Please refer to this page
+# (https://www.notion.so/scalingooriginal/New-Nginx-Version-936b92e173a74f5ab0cd59e5155a4880)
+# on how to use it.
+#
+
+set -e
+
+basedir="$( cd -P "$( dirname "$0" )" && pwd )"
+source "$basedir/lib/utils"
+source "$basedir/lib/s3"
+source "$basedir/../config/versions.sh"
+
+# Binding between modsecurity and nginx (since modsecurity can be used as standalone or with apache for instance)
+MODSECURITY_NGINX_VERSION="v1.0.2"
+MODSECURITY_NGINX_DIR="modsecurity-nginx-${MODSECURITY_NGINX_VERSION}"
+MODSECURITY_NGINX_TARBALL="${MODSECURITY_NGINX_DIR}.tar.gz"
+MODSECURITY_NGINX_URL="https://github.com/SpiderLabs/ModSecurity-nginx/releases/download/${MODSECURITY_NGINX_VERSION}/${MODSECURITY_NGINX_TARBALL}"
+
+nginx_version="${1:-$default_nginx_version}"
+zlib_version="$default_zlib_version"
+
+validate_env
+
+export PATH=${basedir}/../vendor/bin:$PATH
+
+if [ -z "$nginx_version" ]; then
+  echo "Usage: $0 <version>" >&2
+  exit $E_ARG_MISSING
+fi
+
+if [ -z "$NGINX_PCRE_VERSION" ]; then
+  # From https://sourceforge.net/projects/pcre/files/pcre/
+  NGINX_PCRE_VERSION=8.45
+fi
+
+pcre_version="$NGINX_PCRE_VERSION"
+
+tempdir=$(mktmpdir nginx)
+cd $tempdir
+
+echo "-----> Downloading dependency PCRE ${pcre_version}"
+
+curl -LO "https://sourceforge.net/projects/pcre/files/pcre/${pcre_version}/pcre-${pcre_version}.tar.gz"
+tar -xzvf "pcre-${pcre_version}.tar.gz"
+
+echo "-----> Downloading dependency zlib ${zlib_version}"
+
+s3_download "zlib/zlib-${zlib_version}.tar.gz"
+tar -xzvf "zlib-${zlib_version}.tar.gz"
+
+if [ "NO_MODSECURITY_MODULE" != "true" ] ; then
+  echo "---> Installing modsecurity-nginx"
+  curl --remote-name --location "$MODSECURITY_NGINX_URL"
+  tar --extract --ungzip --verbose --file "$MODSECURITY_NGINX_TARBALL"
+  nginx_config_dir="/app/vendor/nginx/etc/nginx"
+  mkdir -p "${nginx_config_dir}/modules" "${nginx_config_dir}/modsec"
+  # cp objs/ngx_http_modsecurity_module.so "${nginx_config_dir}/modules"
+  # cp ../${MODSECURITY_DIR}/unicode.mapping "${nginx_config_dir}/modsec"
+fi
+
+echo "-----> Downloading NGINX ${nginx_version}"
+
+curl -LO "https://nginx.org/download/nginx-${nginx_version}.tar.gz"
+tar -xzvf "nginx-${nginx_version}.tar.gz"
+
+echo "-----> Uploading source to build server"
+
+pushd nginx-${nginx_version} >/dev/null && \
+  ./configure \
+    --prefix=/app/vendor/nginx \
+    --with-http_ssl_module \
+    --with-http_auth_request_module \
+    --with-pcre=../pcre-${pcre_version} \
+    --with-zlib=../zlib-${zlib_version} \
+    "--add-dynamic-module=../${MODSECURITY_NGINX_DIR}" \
+    --with-http_realip_module && \
+  make && make install && \
+popd > /dev/null
+
+echo "-----> Uploading package to S3 ${S3_URL}/package/nginx-${nginx_version}.tgz"
```

Comment:
Currently Nginx is packaged in the PHP buildpack, and it's uploaded to OVH object storage. Is there a plan to deprecate the use of OVH in favor of S3? Should we update the PHP buildpack accordingly?

Reply:
I think that yes, we should build a plan for this, I can do it

Reply:
https://github.com/Scalingo/php-buildpack/issues/240

Reply:
https://github.com/Scalingo/semver.io/issues/22

Reply:
Currently all versions of nginx available through PHP buildpacks have been built in the new S3 bucket, thus it's retrocompatible